### PR TITLE
feat(filters): add lower level support for filters

### PR DESCRIPTION
**Pull request by tkrugg**

### CODE REVIEW

File: lib/_sendEvent.ts [hunk 1]
```diff
@@ -73,14 +79,30 @@ export function sendEvent(
...
+    }
+    if (!Array.isArray(eventData.filters)) {
+      throw TypeError("expected optional parameter `filters` to be an array");
+    }
+    event.filters = eventData.filters;
+  }
+
+  if (isUndefined(eventData.objectIDs) && isUndefined(eventData.filters)) {
+    throw new Error("expected either `objectIDs` or `filters` to be provided");
+  }
```
Comment (samouss) [inline lib/_sendEvent.ts]: The function became a bit dense. In the future if more parameter are added it will become hard to follow IMO. A possible solution would be to externalise all the validation functions and perform them as soon as possible. It has the advantage to be focused on each method. The downside is that we have to repeat the same validation for the common parameters but the tradeoff seems acceptable to me. WDYT @tkrugg? We can do it later though no problem at the moment.

Comment (tkrugg) [reply lib/_sendEvent.ts]: I agree. 
I've opened an issue for it here: https://github.com/algolia/search-insights.js/issues/44
